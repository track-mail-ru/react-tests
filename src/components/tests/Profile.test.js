import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Profile from '../Profile';

const mockStore = configureStore([]);

describe('Test Redux Components', () => {
	let component;
	let store;

	beforeEach(() => {
		store = mockStore({
			globalState: {state: {frameStyles: {Profile: {}}}},
		});
		component = renderer.create(
			<Provider store={store}>
				<Profile />
			</Provider>,
		);
	});

	it('renders somehow', () => {
		expect(component.toJSON()).toMatchSnapshot();
	});
});
