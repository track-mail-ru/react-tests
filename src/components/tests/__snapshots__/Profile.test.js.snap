// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Redux Components renders somehow 1`] = `
<div
  className="conteiner"
  style={Object {}}
>
  <div
    className="header"
  >
    <div
      className="menu"
    />
    <span
      className="formName"
    >
      Профиль
    </span>
  </div>
  <div
    className="content"
  >
    <div
      className="topInfo"
    >
      <div
        className="photo"
      >
        <div
          className="userConteiner"
        >
          <div
            className="image"
            style={
              Object {
                "backgroundImage": "url(image.jpg)",
                "backgroundPosition": "cetner center",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "cover",
              }
            }
          />
          <div
            className="loginConteiner"
          >
            <div
              className="fullName"
            >
              Виталий Кисель
            </div>
            <div
              className="status"
            >
              онлайн
            </div>
          </div>
        </div>
        <div
          className="photoButton"
        >
          <div
            className="phImage"
          />
        </div>
      </div>
      <div
        className="info"
      >
        <div
          className="InfoInput"
        >
          <div
            className="inputBlock active"
          >
            <div
              className="placeholder"
            >
              Никнейм
            </div>
            <input
              className="input"
              defaultValue="@jellyb0y"
              onChange={[Function]}
            />
          </div>
        </div>
        <div
          className="InfoInput"
        >
          <div
            className="inputBlock active"
          >
            <div
              className="placeholder"
            >
              Почта
            </div>
            <input
              className="input"
              defaultValue="test@mail.ru"
              onChange={[Function]}
            />
          </div>
        </div>
        <div
          className="InfoInput"
        >
          <div
            className="inputBlock active"
          >
            <div
              className="placeholder"
            >
              О себе
            </div>
            <input
              className="input"
              defaultValue="Разработчик этого приложения"
              onChange={[Function]}
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
